handlers.go

```diff
@@ -4302,3 +4305,53 @@ func (s *Server) handleRunGc(c echo.Context) error {
 
 	return nil
 }
+
+func (s *Server) handleGateway(c echo.Context) error {
+	proto, cc, segs, err := gateway.ParsePath("/" + c.Param("path"))
+	if err != nil {
+		return err
+	}
+
+	redir, err := s.checkGatewayRedirect(proto, cc, segs)
+	if err != nil {
+		return err
+	}
+
+	if redir == "" {
+		s.gwayHandler.ServeHTTP(c.Response().Writer, c.Request())
+		return nil
+	}
+
+	return c.Redirect(302, redir)
```

Comment:
Actually 302 is incorrect here, we should always 307 ( meaning: as a UA never store the destination, even for temporary sessions. Always come to estuary.tech first )
```suggestion
	return c.Redirect(307, redir)
```